internal/rpmmd/repository.go

```diff
@@ -375,3 +378,45 @@ func (packages PackageList) ToPackageInfos() []PackageInfo {
 
 	return results
 }
+
+func (r RepoConfig) MarshalJSON() ([]byte, error) {
+	type aliasType RepoConfig
+	type compatType struct {
+		aliasType
+
+		BaseURL string `json:"Baseurl,omitempty"`
+	}
+	compatRepo := compatType{
+		aliasType: aliasType(r),
+	}
+
+	var baseUrl string
+	if len(r.BaseURLs) > 0 {
+		baseUrl = strings.Join(r.BaseURLs, ",")
+	}
+
+	compatRepo.BaseURL = baseUrl
+
+	return json.Marshal(compatRepo)
+}
+
+func (r *RepoConfig) UnmarshalJSON(data []byte) error {
+	type aliasType RepoConfig
+	type compatType struct {
+		aliasType
+
+		BaseURL string `json:"Baseurl,omitempty"`
+	}
+
+	var compatRepo compatType
+	if err := json.Unmarshal(data, &compatRepo); err != nil {
+		return err
+	}
+
+	if compatRepo.BaseURL != "" {
+		compatRepo.BaseURLs = strings.Split(compatRepo.BaseURL, ",")
```

Comment:
Ah  I missed this, I've left it as a list since the `yum_repo_stage` takes as a list of baseurls as an option.
Happy to change that though. But not sure if it makes a huge difference.